napari/components/_layer_slicer.py

```diff
@@ -216,22 +225,35 @@ def _slice_layers(self, requests: Dict) -> Dict:
         dict[Layer, SliceResponse]: which contains the results of the slice
         """
         result = {layer: request() for layer, request in requests.items()}
-        self.events.ready(Event('ready', value=result))
+        self.events.ready(value=result)
         return result
 
     def _on_slice_done(self, task: Future[Dict]) -> None:
         """
         This is the "done_callback" which is added to each task.
         Can be called from the main or slicing thread.
         """
-        if not self._try_to_remove_task(task):
+        if self._try_to_remove_task(task):
+            self._update_status()
+        else:
             logger.debug('Task not found')
             return
 
         if task.cancelled():
             logger.debug('Cancelled task')
+            self._update_status()
             return
 
+    def _update_status(self):
+        if self._layers_to_task:
+            self._busy = True
+        else:
+            self._busy = False
+
+    @property
+    def busy(self):
+        return self._busy
```

Comment:
Instead of storing extra state (i.e. `_busy`) could we just check `len(self._layers_to_task) > 0`? In that case, we should probably acquire the associated lock in case we're in `_on_slice_done` in the slicing thread.